File: 349-Intersection-of-Two-Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        
        set<int> my_set;
        for (int i = 0; i < nums1.size(); i++) {

            int elem = nums1[i];
            for (int j = 0; j < nums2.size(); j++) {
                if (elem == nums2[j]) {
                    nums2[j] = -1;
                    // arr.push_back(elem);
                    my_set.insert(elem);
                    break;
                }
            }
        }
        vector<int> arr(my_set.begin(),my_set.end());
        return arr;
    }
};

File: 349-Intersection-of-Two-Arrays.java
import java.util.*;
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashMap<Integer,Integer> map2=new HashMap<>();
        HashMap<Integer,Integer> map1=new HashMap<>();
        
        for(int it:nums1){
            map1.put(it,map1.getOrDefault(it,0)+1);
        }
        for(int it:nums2){
            map2.put(it,map2.getOrDefault(it,0)+1);
        }

        ArrayList<Integer> list=new ArrayList<>();

        
        
        for(HashMap.Entry<Integer,Integer> it:map1.entrySet()){
            if(map2.containsKey(it.getKey())==true){
                list.add(it.getKey());
            }
        }
        int arr[] = new int[list.size()];
        int index=0;
        for(int it:list){
            arr[index++]=it;
        }

        return arr;
    }
}